app/lucy/src/app/components/base-form/waypoint/waypoint-modal/waypoint-modal.component.ts

```diff
@@ -0,0 +1,357 @@
+      this.waypointEntryComponents.push(new WaypointTextEntryComponent(this.validation, this.converter));
+      this.waypointEntryComponents.push(new WaypointTextEntryComponent(this.validation, this.converter));
+      this.maxPointsLengthReached = this.waypointEntryComponents.length === this.MAX_NUM_POINTS ? true : false;
+    }
+  }
+
+  toggleView() {
+    this.showInfo = !this.showInfo;
+  }
+
+  toggleUTM() {
+    this.showLatLong = false;
+  }
+
+  toggleLAT() {
+    this.showLatLong = true;
+  }
+
+  offsetChanged(value: number) {
+    this.offset = value;
+    this.inputDirtyFlag = true;
+  }
+
+  pointChanged(coordinate: LatLongCoordinate, index: number) {
+    this.points.splice(index, 1, coordinate);
+    this.inputDirtyFlag = true;
+  }
+
+  geoJSONChangeHandler(json: any) {
+    this.geoJSON = json;
+    this.geoJSONEmitter.emit(this.geoJSON);
+  }
+
+  /**
+   * parses points from input GeoJSON and adds them to this.points
+   */
+  private addPointsFromGeoJSON() {
+    for (const feature of this.geoJSON['features']) {
+      if (feature['geometry']['type'] === 'Polygon') {
+        this.offset = feature['properties']['offset'];
+      } else if (feature['geometry']['type'] === 'Point') {
+        this.points.push({latitude: feature['geometry']['coordinates'][1], longitude: feature['geometry']['coordinates'][0]});
+      }
+    }
+  }
+
+  private createWaypointTextEntriesForPoints() {
+    // first remove any existing waypoint entry components
+    this.waypointEntryComponents = [];
+    for (const pt of this.points) {
+      const wte = new WaypointTextEntryComponent(this.validation, this.converter);
+      wte.point = pt;
+      this.waypointEntryComponents.push(wte);
+      this.maxPointsLengthReached = this.waypointEntryComponents.length === this.MAX_NUM_POINTS ? true : false;
+    }
+  }
+
+  addNewWaypointTextEntry() {
+    const newWaypointEntry = new WaypointTextEntryComponent(this.validation, this.converter);
+    this.waypointEntryComponents.push(newWaypointEntry);
+    this.maxPointsLengthReached = this.waypointEntryComponents.length === this.MAX_NUM_POINTS ? true : false;
+    this.points.push(newWaypointEntry.point);
+    this.inputDirtyFlag = true;
+  }
+
+  removeWaypoint(index: number) {
+    this.waypointEntryComponents.splice(index, 1);
+    this.points.splice(index, 1);
+    this.maxPointsLengthReached = this.points.length === this.MAX_NUM_POINTS ? true : false;
+    this.inputDirtyFlag = true;
+  }
+
+  waypointEntryDone() {
+    this.geoJSONEmitter.emit(this.geoJSON);
+    this.onClose.emit({offset: this.offset, points: this.points, polygon: this.polygon});
+  }
+
+  generatePath() {
+    this.errors = [];
+    if (!this.validWaypointValues(this.offset, this.points)) {
+      this.errors.push('Error: please fill all the fields');
+      return;
+    }
+
+    if (this.waypointEntryComponents.length > this.MAX_NUM_POINTS) {
+      this.errors.push('Error: cannot enter more than 10 coordinates');
```

Comment:
This should be `more than 20 coordinates`

Reply:
Fixed. (I removed the hard-coded min/max values here)